cmd/queue/main.go

```diff
@@ -162,8 +162,10 @@ func main() {
 	}()
 
 	// Setup probe to run for checking user-application healthiness.
+	// Do not set up probe if concurrency state endpoint is set, as
+	// paused containers don't play well with k8s readiness probes.
 	probe := func() bool { return true }
-	if env.ServingReadinessProbe != "" {
+	if env.ServingReadinessProbe != "" && env.ConcurrencyStateEndpoint == "" {
```

Comment:
How about putting this behind a flag since you are not 100% convinced with it, even if enabled by default, but at least allows users to turn it off if they want to?